### PR TITLE
Fix warnings in verify_params

### PR DESCRIPTION
**Pull request by lcjohnso**
In #460, @jwuphysics swapped `exit()` interrupts with errors and warnings.  The non-interrupting warnings, which were previously handled by the `noexit` option, were not actually implemented as warnings but as interrupting errors.

This PR swaps the non-existent `UserWarning` with a `warnings.warn()` call (whose default type is UserWarning).

### CODE REVIEW

**Comment by jwuphysics** [comment]
Looks good!

**Comment by karllark** [comment]
One question, will the new behavior be to issue a warning or an error?  In other words, does execution continue or stop?  I'm asking as it sounds like execution will now continue and this would be a change from the previous behavior.  Just want to make sure I understand the change fully.

**Comment by jwuphysics** [comment]
Ah good point, sorry I totally missed that! `raise` will exit the program (similar to the original behavior) but `warnings.warn()` will continue. So in fact the behavior will change.

**Comment by lcjohnso** [comment]
The change I'm trying to reverse is the loss of the `noexit` option -- an ignorable, non-interrupting version of these warnings was removed.  As valid cases exists where `fA` is not used by the extinction model, or grid params are single valued, the current forced behavior of throwing an interrupting error is not acceptable.

It seems like "best of both worlds" would be that the current warnings throw an interrupting error by default, but that they can be set to ignorable.  That can be accomplished by using `warnings.simplefilter('error', UserWarning)` by default, which turns warnings into errors, but provide an `allow_warnings` option that would skip the `simplefilter` call and make the warnings  non-interrupting.  Essentially reverting the loss of the `noexit`-like option that was removed by #460...

@karllark & @jwuphysics: if this is a reasonable compromise, I'll implement this.

**Comment by karllark** [comment]
This sounds like a great solution.  Go for it.

**Comment by jwuphysics** [comment]
I agree with Karl. Thanks for catching the change in behavior.

**Comment by lcjohnso** [comment]
New implementation is in place: default behavior is to raise exception, but if `allow_warnings` is set to `True` in datamodel.py, then warnings become non-interrupting.

I need to add a test: if `fA=None` in datamodel.py, then raise UserWarning exception during `verify_parameters.verify_input_format()` by default.  If `datamodel.allow_warnings` is `True`, then raise non-interrupting warning during `verify_parameters.verify_input_format()` and continue.

Help on implementing this test would be much appreciated!

**Comment by karllark** [comment]
A test would be awesome!

Have you looked at the existing tests for examples?  Or are they not sufficient?  There are examples of tests that look for warnings/errors in the dust_extinction package if more examples would be useful.  I can also just dig out probably what you need.

**Comment by lcjohnso** [comment]
I'm stuck on writing a test for this.  My idea was to model a couple tests from [dust_extinction/tests/test_warnings.py](https://github.com/karllark/dust_extinction/blob/master/dust_extinction/tests/test_warnings.py), but the fact that verify_params.verify_input_format() takes `datamodel` module as input means I'd really need to create test versions (e.g., datamodel_test_noallow.py, datamodel_test_allow.py), either in beast/tools/tests/ or externally, that could be imported.  Yuck.

In the meantime, I found and corrected a bug by feeding verify_params.verify_input_format() various test datamodels -- I wish there was an easier way to capture that process in a coded test.  Suggestions welcome!

**Comment by karllark** [comment]
datamodel.py could be mocked up as a class.  This is how it behaves.  Could this solve  the problem?

**Comment by lcjohnso** [comment]
Yes, I think that would do it.  I'll give this a shot and ping again when I've got code to review.

**Comment by lcjohnso** [comment]
@karllark Tests are in place and this PR is ready for review and merge.  Looks like the Codacy CI tool should be tuned so it doesn't complain about `assert` calls in tests.

File: beast/tools/tests/test_verify_warnings.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    """Mock datamodel w/ single-valued R_V and allow_warnings = True"""
+    allow_warnings = True
+
+def test_verifyparams_errorRV():
+    """Test: verify_params for case of warning raising exception."""
+    with pytest.raises(UserWarning) as exc:
+        verify_params.verify_input_format(datamodel_mock_RV())
+    assert exc.value.args[0] == "Note: rvs grid is single-valued."
+
+def test_verifyparams_errorRV():
```
Comment (karllark) [inline beast/tools/tests/test_verify_warnings.py]: Duplicate function name.  Should be different.  Or is this just a duplication of the entire function?

**Comment by karllark** [comment]
Working on removing the codacy check of assert statements.  I agree this is annoying.

File: beast/tools/tests/test_verify_warnings.py (hunk 1 above)
Comment (jwuphysics) [reply beast/tools/tests/test_verify_warnings.py]: Duplicate function name: the two functions differ as one verifies `datamodel_mock_RV()` and the other `datamodel_mock_noallowRV()`.

Comment (lcjohnso) [reply beast/tools/tests/test_verify_warnings.py]: Thanks for catching.  This was me wanting to add one more case to test; I'll edit function name and docstring, so should be good to go.

**Comment by lcjohnso** [comment]
> Working on removing the codacy check of assert statements. I agree this is annoying.

Great -- would rather have fewer CI formatting checks than start ignoring PR check failures.

**Comment by karllark** [comment]
Great PR!  It has code, tests, and docs.  Kudos @lcjohnso!
